pkg/imgpkg/imageset/image_set.go

```diff
@@ -77,57 +79,135 @@ func (o *ImageSet) Import(imgOrIndexes []imagedesc.ImageOrIndex,
 	o.logger.WriteStr("importing %d images...\n", len(imgOrIndexes))
 	defer func() { o.logger.WriteStr("imported %d images\n", len(importedImages.All())) }()
 
-	errCh := make(chan error, len(imgOrIndexes))
 	importThrottle := util.NewThrottle(o.concurrency)
 
+	imageOrIndexesToWrite := map[regname.Reference]regremote.Taggable{}
+	var imageOrIndexesToWriteLock = &sync.Mutex{}
+	errCh := make(chan error, len(imgOrIndexes))
 	for _, item := range imgOrIndexes {
 		item := item // copy
 
 		go func() {
 			importThrottle.Take()
 			defer importThrottle.Done()
+			addImageOrImageIndexToTaggableMapForMultiWrite(item, errCh, importRepo, registry, imageOrIndexesToWrite, imageOrIndexesToWriteLock)
+		}()
+	}
 
-			existingRef, err := regname.NewDigest(item.Ref())
-			if err != nil {
-				errCh <- err
-				return
-			}
+	err := checkForAnyAsyncErrors(imgOrIndexes, errCh)
+	if err != nil {
+		return nil, err
+	}
 
-			importDigestRef, err := o.importImage(item, existingRef, importRepo, registry)
-			if err != nil {
-				errCh <- fmt.Errorf("Importing image %s: %s", existingRef.Name(), err)
-				return
-			}
+	err = registry.MultiWrite(imageOrIndexesToWrite, o.concurrency)
```

Comment:
it seems that MultiWrite is only going to work against same repo [0] (is that going to screw us up if we decide to do copy with multiple repo destinations based on where user wants us to write images to @joaopapereira)

[0] https://github.com/google/go-containerregistry/blob/4d068fbcb51f224f9667497fe24b83506f979ede/pkg/v1/remote/multi_write.go#L34

Reply:
@cppforlife and I had a convo, we think that in the future, if setting custom destination renaming of images, we are not blocked technically by this change. We could for e.g. call MultiWrite, multiple times.